classifiers/nlp/ZeroShotTFClassifier/__init__.py

```diff
@@ -0,0 +1,236 @@
+
+        if self.pooling_strategy == "auto":
+            self.pooling_strategy = "cls"
+            self.logger.warning(
+                '"auto" pooling_strategy is deprecated, Defaulting to '
+                ' "cls" to maintain the old default behavior.'
+            )
+
+        if self.pooling_strategy not in ["cls", "mean", "max", "min"]:
+            self.logger.error(
+                f"pooling strategy not found: {self.pooling_strategy}."
+                ' The allowed pooling strategies are'
+                '"cls", "mean", "max", "min".'
+            )
+            raise NotImplementedError
+
+        if len(self.labels) < 2:
+            raise ValueError("The number of target labels must be at least 2.")
+
+        if len(self.labels) != \
+                len(set(self.labels)):
+            raise ValueError(
+                "There are duplicate value in the target_label argument."
+            )
+
+    @batching
+    @as_ndarray
+    def predict(self,
+                data: "np.ndarray",
+                *args,
+                **kwargs) -> "np.ndarray":
+        """
+         Perform zero shot classification on `data`, the predicted label
+         for each sample in X is returned.
+
+         The output is a zero/one one-hot label for L-class multi-class
+         classification of size (B, L) with `B` being `data.shape[0]`
+         and `L` being the number of potential classification labels.
+
+         :param data: the input textual data to be classified, a 1 d
+            array of string type in size `B`
+         :type data: np.ndarray
+         :return: zero/one one-hot predicted label of each sample
+            in size `(B, L)`
+         :rtype: np.ndarray
+        """
+
+        data_encoded = self._encode(data)
+        labels_encoded = self._encode(self.labels)
+
+        distances = self._evaluate(data_encoded, labels_encoded)
+
+        labels_pred = (distances == distances.min(axis=1)[:, None]).astype(int)
+
+        return labels_pred
+
+    def post_init(self):
+        from transformers import TFAutoModel, AutoTokenizer
+
+        self.tokenizer = AutoTokenizer.from_pretrained(
+            self.base_tokenizer_model
+        )
+        self.model = TFAutoModel.from_pretrained(
+            self.pretrained_model_name_or_path, output_hidden_states=True
+        )
+        self.to_device()
+
+    def _evaluate(self,
+                  actual: "np.ndarray",
+                  desired: "np.ndarray") -> "np.ndarray":
+
+        actual = _expand_vector(actual)
+        desired = _expand_vector(desired)
+
+        return _cosine(_ext_A(_norm(actual)), _ext_B(_norm(desired)))
+
+    def _encode(self, data: "np.ndarray", *args, **kwargs) -> "np.ndarray":
+
+        import tensorflow as tf
+
+        if not self.tokenizer.pad_token:
+            self.tokenizer.add_special_tokens({"pad_token": "[PAD]"})
+            self.model.resize_token_embeddings(len(self.tokenizer.vocab))
+
+        input_tokens = self.tokenizer(
```

Comment:
are we sure we are handling properly the input if the model works in GPU?

Reply:
In the requirement, it is specified to use `tensorflow-cpu` and therefore we don't need to consider GPU usages.

Reply:
but as many other executors, but considering it can help us to prepare for its usage.

Plus some users may prefer to install their dependencies and use it without dockee